pkg/genericapiserver/config.go

```diff
@@ -354,50 +352,48 @@ func (c completedConfig) New() (*GenericAPIServer, error) {
 	}
 
 	s.installAPI(c.Config)
-	s.Handler, s.InsecureHandler = s.buildHandlerChains(c.Config, http.Handler(s.Mux.BaseMux().(*http.ServeMux)))
+
+	// Wire the handler chains
+	s.Handler = c.BuildHandlerChainFunc(s.HandlerContainer.ServeMux, c.Config, true)
+	s.InsecureHandler = c.BuildHandlerChainFunc(s.HandlerContainer.ServeMux, c.Config, false)
 
 	return s, nil
 }
 
-func (s *GenericAPIServer) buildHandlerChains(c *Config, handler http.Handler) (secure http.Handler, insecure http.Handler) {
-	// filters which insecure and secure have in common
-	handler = genericfilters.WithCORS(handler, c.CorsAllowedOriginList, nil, nil, nil, "true")
+func DefaultBuildHandlerChain(handler http.Handler, c *Config, secure bool) http.Handler {
```

Comment:
Why change this signature?  Why not have one function that returns both and the provider gets to choose how to integrate the chains